core/src/interfaces.rs

```diff
@@ -202,8 +242,12 @@ pub async fn cancel_proof(
 pub struct ProofRequest {
     /// The block number for the block to generate a proof for.
     pub block_number: u64,
+    /// The block number for the block to generate a proof for.
+    pub batch_id: u64,
     /// The l1 block number of the l2 block be proposed.
     pub l1_inclusion_block_number: u64,
+    /// To support batch proof generation.
+    pub l2_block_numbers: Vec<u64>,
```

Comment:
The `ProofRequest.l2_block_numbers` input from the user's HTTP interface is not actually used, because the reqactor backend will obtain the real l2 block numbers by [`parse_l1_batch_proposal_tx_for_pacaya_fork`](https://github.com/taikoxyz/raiko/pull/456/files#diff-dcbc7e4a44885478ec09bb1109302f660ccb5aa04faa1871544fa5e91b74a118R668), and subsequent code also uses the parsed l2 block numbers.

So I think we should remove `l2_block_numbers` from `ProofRequest`.

Reply:
yes, can be removed